File: data/src/main/java/home/oleg/placesnearme/model/DetailedVenueDbEntity.java
package home.oleg.placesnearme.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

@Entity(tableName = "detailed_venue")
public class DetailedVenueDbEntity {

    @PrimaryKey
    @NonNull
    private String id;
    private String title;
    private String address;
    private double lat;
    private double lng;
    private String description;
    private String openingHoursStatus;
    private String formattedPhone;
    private float rating;

    @NonNull
    public String getId() {
        return id;
    }

    public void setId(@NonNull String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public double getLat() {
        return lat;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public double getLng() {
        return lng;
    }

    public void setLng(double lng) {
        this.lng = lng;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getOpeningHoursStatus() {
        return openingHoursStatus;
    }

    public void setOpeningHoursStatus(String openingHoursStatus) {
        this.openingHoursStatus = openingHoursStatus;
    }

    public String getFormattedPhone() {
        return formattedPhone;
    }

    public void setFormattedPhone(String formattedPhone) {
        this.formattedPhone = formattedPhone;
    }

    public float getRating() {
        return rating;
    }

    public void setRating(float rating) {
        this.rating = rating;
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/data/model/DetailedVenueDbEntity.kt
package home.oleg.placesnearme.data.model

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import home.oleg.placenearme.models.Category
import home.oleg.placenearme.models.Contact
import home.oleg.placenearme.models.Location

@Entity(tableName = "detailed_venue")
data class DetailedVenueDbEntity(
        @PrimaryKey
        val id: String,
        val title: String,
        val description: String?,
        val rating: Float,
        val isFavorite: Boolean,
        val isHereNow: Boolean,
        @Embedded
        val location: Location,
        @Embedded
        val contact: Contact?,
        @Embedded
        val category: Category?)

